actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -217,7 +217,17 @@ def fragment_name_with_digest(name, virtual_path) #:nodoc:
 
       # TODO: Create an object that has caching read/write on it
       def fragment_for(name = {}, options = nil, &block) #:nodoc:
-        read_fragment_for(name, options) || write_fragment_for(name, options, &block)
+        if has_log_payload?
+          log_payload[:cache_partial] = true
+          log_payload[:cache_hit] = false
+        end
+
+        if content = read_fragment_for(name, options)
+          log_payload[:cache_hit] = true if has_log_payload?
+          content
+        else
+          write_fragment_for(name, options, &block)
+        end
```

Comment:
I think this would be simpler if we just had the branches mirror each other:

```ruby
if content = read_fragment_for(name, options)
  log_payload[:cache_hit] = true if respond_to?(:log_payload)
  content
else
  log_payload[:cache_hit] = false if respond_to?(:log_payload)
  write_fragment_for(name, options, &block)
end
```

(We don't need to check that `log_payload` is a hash, let's just duck-type it.)